crates/wasm-smith/src/core/code_builder.rs

```diff
@@ -4626,6 +4625,11 @@ fn memory_offset(u: &mut Unstructured, module: &Module, memory_index: u32) -> Re
 
     let choice = u.int_in_range(0..=a + b + c - 1)?;
     if choice < a {
+        let min = module
+            .config
+            .reserved_memory_size()
+            .map(|reserved| min.saturating_sub(reserved).saturating_sub(16))
```

Comment:
why do you need `sub(16)`?

Reply:
see that this function does. it does exactly same above. it subtract 16 bytes for u128 types